### PR TITLE
COMMON: Add initial config file option.

### PR DESCRIPTION
**Pull request by elasota**
This allows an initial/default configuration file to be specified via the command line.  This allows a default settings file to be bundled and loaded even if it's installed to a read-only location, such as Program Files on Windows, and allows the written config file to be deleted without losing the custom defaults.

The way this works is that the config file specified with -i or --initial-cfg is loaded only if the config file is missing from its usual location (i.e. on initial startup, or after the user deleted their config file).

Basically, this is designed to make it easier for things like re-releases to bundle their own default settings with the game, without the drawbacks of setting via the command line (no way for user to change any forced settings, not all options available), or installing default config to a writable location and using --config (more complicated deployment, only way to restore "default" settings is to reinstall).

### CODE REVIEW

**Comment by sev-** [comment]
Thank you!

**Comment by lephilousophe** [comment]
This breaks Dreamcast build: https://buildbot.scummvm.org/#/builders/95/builds/6471
